### PR TITLE
Increase JupyterHub security

### PR DESCRIPTION
**Pull request by tlvu**
Read each commit comments for more infos.

### CODE REVIEW

**Comment by tlvu** [comment]
@davidcaron Please leave comments if you have any.  I just had to merge this PR quickly so I can go-live on our prod instance.

**Comment by tlvu** [comment]
Went live on prod Boreas:
```
deploy finished START_TIME=2020-02-06T15:25:35-0500
deploy finished   END_TIME=2020-02-06T15:28:29-0500
```

File: birdhouse/config/jupyterhub/jupyterhub_config.py.template [hunk 1]
```diff
@@ -62,6 +62,9 @@ def create_dir_hook(spawner):
 
 c.Spawner.pre_spawn_hook = create_dir_hook
 
+## Disable per-user configuration of single-user servers.
+c.Spawner.disable_user_config = True
```
Comment (davidcaron) [inline birdhouse/config/jupyterhub/jupyterhub_config.py.template]: Not sure, but I think this has no effets when using dockerspawner?

File: birdhouse/config/jupyterhub/jupyterhub_config.py.template [hunk 2]
```diff
@@ -67,4 +67,7 @@ c.JupyterHub.log_level = logging.DEBUG
 
 c.Spawner.debug = True
 
+## Extra arguments to be passed to the single-user server.
+c.Spawner.args = ["--NotebookApp.terminals_enabled=False"]
```
Comment (davidcaron) [inline birdhouse/config/jupyterhub/jupyterhub_config.py.template]: The terminal is disabled, but the user can still launch bash commands from a notebook... If the terminal is not used, you're right that it should be disabled.

File: birdhouse/config/jupyterhub/jupyterhub_config.py.template (hunk 1 above)
Comment (tlvu) [reply birdhouse/config/jupyterhub/jupyterhub_config.py.template]: Apparently user can influence how the jupyter user server starts with stuff in HOME/.jupyter, and we persist HOME now ... so I'd rather be on the cautious side.  But agree by using dockerspawner, the risk is already reduced or this setting might have no effect.  Was following the doc in the commit message.

File: birdhouse/config/jupyterhub/jupyterhub_config.py.template (hunk 2 above)
Comment (tlvu) [reply birdhouse/config/jupyterhub/jupyterhub_config.py.template]: You're right, this just make it less obvious to get to the terminal and we have no use-case for the terminal yet.

**Comment by tlvu** [comment]
Tagged `1.8.1`.

**Pull request merged**